sklearn/utils/tests/test_array_api.py

```diff
@@ -164,6 +168,33 @@ def test_asarray_with_order_ignored():
     assert not X_new_np.flags["F_CONTIGUOUS"]
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "sample_weight, normalize, expected",
+    [
+        (None, False, 10.0),
+        (None, True, 2.5),
+        ([0.4, 0.4, 0.5, 0.7], False, 5.5),
+        ([0.4, 0.4, 0.5, 0.7], True, 2.75),
+        ([1, 2, 3, 4], False, 30.0),
+        ([1, 2, 3, 4], True, 3.0),
+    ],
+)
+def test_weighted_sum(
+    array_namespace, device, dtype, sample_weight, normalize, expected
+):
+    xp, device, _ = _array_api_for_tests(array_namespace, device, dtype)
+    sample_score = xp.asarray([1, 2, 3, 4])
```

Comment:
In addition to the previous comment, I think we should explicitly convert `sample_weight` as well (see also #27232 for my thinking on this)
```suggestion
    sample_score = xp.asarray([1, 2, 3, 4])
    sample_weight = xp.asarray(sample_weight, device=device, dtype=dtype)
```